Sources/AudioKit/MIDI/MIDIInstrument.swift

```diff
@@ -15,7 +15,7 @@ open class MIDIInstrument: PolyphonicNode, MIDIListener, NamedNode {
     open var midiIn = MIDIEndpointRef()
 
     /// Name of the instrument
-    open var name = "AudioKit MIDI Instrument"
+    open var name: String? = "AudioKit MIDI Instrument"
```

Comment:
I don't agree with making `name` optional. Since a default is provided, this would only be `nil` if the user expressely _removed_ the name and set it to `nil`, which doesn't seem like a case that should be supported or, at the very least, encouraged. Optionals should only be used when there's a good reason for them, and I don't see any good reason here.

Reply:
The intent here was to avoid the `connectionTreeDescription` printing something like `Mixer ("Mixer")` if i were to provide `"Mixer"` as the default. The `connectionTreeDescription` (which i cannot thank you enough for getting the ball rolling on), prints the type by default, and the name if provided. Giving a default to the `Mixer` class would be fine, but do you think something like `"Mixer"` would be a reasonable default? That would then remove the optional name, and make it required.

Reply:
Yeah, I feel you on that concern about stuff like `Mixer("Mixer")` but I think that's a pretty small price to pay, and it's really just a debugging tool anyway. The existing other `NamedNode` types have the same behavior as a result of their defaults, so it's not a step backwards. The other thing that gives me pause about making this optional is that `MIDIDestinationCreateWithBlock` which is clearly expecting some name to be provided. The ! force-unwrap will definitely fail as soon as one of these is set to `nil`, since ! means "I promise this won't be `nil` when you use it".

Reply:
The is another possible approach we could think about here, which is to change the behavior of connectionTreeDescription to _only_ show the name if the node is a `NamedNode`. That would allow the defaults to be useful by themselves. The only concern I have about that is that it might make some named values be confusing since it would then hide their type, but I guess that would be up to the developer to avoid. For example, if a `Mixer` was named "1", the developer would have to understand that it would be better to name it "Mixer 1" or something like that if they want the tree to be readable.

What do you think about that alternative? It seems like it solves all the problems.

I'd like to know what @aure thinks about all this, too. I'm not actually an official AudioKit team member, and these are just my opinions.

Reply:
https://github.com/AudioKit/AudioKit/compare/v5-develop...hoopes:mixer-name-redux?expand=1

What do you think of this instead?

Reply:
I think that's better, but I would not recommend having the name parameters be optional with defaults of nil in the initializers. If you make them non-optional but give them good defaults, then existing code can still ignore their addition. I would recommend modeling the approach closely around what's done in `MIDISampler`.

Your thoughts on this have really gotten me thinking that updating `connectionTreeDescription` will make the adoption of `NamedNode` much less painful. I think I'll play with that some, and it may feed into your changes here and make everything much smoother overall. But in the meantime, there's no reason why `Mixer` couldn't adopt naming using the exact same pattern as `MIDISampler`.

Reply:
Regardless of whether you guys are core team members, you are the ones who are using and developing and to some extent advocating for this useful (I like it too though!). What would be kind of cool is if the default name could always be unique. Perhaps the pointer location of the node would be a nice unique identifier to use?